lib/read.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+/* global JuttleAdapterAPI */
+let AdapterRead = JuttleAdapterAPI.AdapterRead;
+let JuttleMoment = JuttleAdapterAPI.types.JuttleMoment;
+let errors = JuttleAdapterAPI.errors;
+let google = require('./google');
+
+class ReadGoogleAnalytics extends AdapterRead {
+    constructor(options, params) {
+        super(options, params);
+
+        if (params.filter_ast) {
+            throw errors.compileError('ADAPTER-UNSUPPORTED-FILTER', {
+                proc: 'read googleanalytics',
+                filter: 'filtering',
+                location: params.filter_ast.location
+            });
+        }
+
+        // to provide better error messages, the optimizer includes information
+        // as to the invalid invocation in the optimization_info
+        if (params.optimization_info.invalid) {
+            // XXX this would be good to move into the core juttle errors
+            throw new errors.CompileError(`read googleanalytics does not support ${params.optimization_info.invalid}`,
+                'ADAPTER-UNSUPPORTED-FEATURE',
+                {
+                    location: params.optimization_info.location
+                }
+            );
+        }
+
+        if (params.optimization_info.type !== 'reduce') {
+            throw new errors.CompileError('read googleanalytics does not support read without reduce',
+                'ADAPTER-INVALID-READ',
+                {
+                    location: params.location
+                }
+            );
+        }
+
+        this.metrics = params.optimization_info.metrics;
+        this.dimensions = params.optimization_info.dimensions;
+        this.metagroup = params.optimization_info.metagroup;
+        this.webProperty = options.webProperty;
+        this.viewId = options.viewId;
+        this.fetchSize = options.fetchSize;
+
+        if (this.options.from) {
+            if (! this.options.from.quantize(JuttleMoment.duration(1, 'd')).eq(this.options.from)) {
```

Comment:
This is a pretty involved `if`, can we get a little `function isRoundDay(moment)` that we'd call here and on line 61?